app/src/main/res/raw/fxr_config.yaml

```diff
@@ -26,4 +26,6 @@ prefs:
   browser.gesture.pinch.in: ''
   browser.gesture.pinch.out.shift: ''
   browser.gesture.pinch.in.shift: ''
-  apz.one_touch_pinch.enabled: false
+  apz.one_touch_pinch.enabled: false,
+  # allows scripts to open WebXR immersive sessions
+  dom.vr.require-gesture: false
```

Comment:
One possibility could be to do it in `SessionUtils.prepareConfigurationPath()`, which copies the configuration file from the assets to the device's filesystem. Basically, the idea would be to append an extra line to the file if needed:

```java
final String REQUIRE_GESTURE_CONFIG = "dom.vr.require-gesture: false";
WidgetManagerDelegate widgetManager = (WidgetManagerDelegate) aContext;

if (widgetManager.isOpenInImmersive()) {
        outputStream.write(REQUIRE_GESTURE_CONFIG.getBytes());
}
```

Another approach could be to have two files in the assets, and copy one or the other depending on the current configuration. Maybe that would be better?

Reply:
I think it's enough with the first approach you mentioned. Adding a line to the stream should make it

Reply:
I've been working on this but it is a bit more complex than I expected because `prepareConfigurationPath()` is called very early, when we have not parsed the incoming Intent yet and so we don't know that we will be in the automatically immersive mode.

Specifically, as a result of `VRBrowserActivity.onCreate()` calling

```java
`((VRBrowserApplication) getApplication()).onActivityCreate(this)`.
```

Reply:
I finally found a way to change the configuration prefs only when needed for launching in immersive mode.

The problem is that we only write the prefs file the first time that we create the browser engine, which happens several times at launch time **before** we have parsed the incoming Intent and we can know in which mode we are launching.

The solution is that `WidgetManagerDelegate.isOpenInImmersive()` internally will check the current Intent, to cover the case where it is being called too early.

We then pass `VRBrowserActivity` as the context to the objects that may initialize the engine, so eventually `SessionUtils.prepareConfigurationPath()` is able to decide whether specific prefs should be added to the configuration file.

For launching Wolvic directly in immersive mode, these prefs are:
```
  dom.vr.require-gesture: false
  media.autoplay.default: 0
```